appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/AppSecDataProvider.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * -
+ * Copyright (C) 2023 Vaadin Ltd
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full license.
+ */
+
+package com.vaadin.appsec.v7.service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.cyclonedx.model.Component;
+import us.springett.cvss.Cvss;
+
+import com.vaadin.appsec.backend.model.osv.response.Affected;
+import com.vaadin.appsec.backend.model.osv.response.OpenSourceVulnerability;
+import com.vaadin.appsec.backend.service.BillOfMaterialsStore;
+import com.vaadin.appsec.backend.service.VulnerabilityStore;
+import com.vaadin.appsec.v7.data.DependencyDTO;
+import com.vaadin.appsec.v7.data.SeverityLevel;
+import com.vaadin.appsec.v7.data.SeverityLevelComparator;
+import com.vaadin.appsec.v7.data.VulnerabilityDTO;
+
+/**
+ * Helper class to provide bill of materials and vulnerabilities as DTOs for use
+ * in the UI.
+ */
+public class AppSecDataProvider {
+
+    /**
+     * Gets dependencies.
+     *
+     * @return all dependencies
+     */
+    public static List<DependencyDTO> getDependencies() {
+        final List<OpenSourceVulnerability> vulnerabilities = VulnerabilityStore
+                .getInstance().getVulnerabilities();
+
+        return BillOfMaterialsStore.getInstance().getBom().getComponents()
+                .stream().map(c -> {
+                    DependencyDTO dep = new DependencyDTO(c.getGroup(),
+                            c.getName(), c.getVersion());
+                    updateVulnerabilityStatistics(dep, vulnerabilities,
+                            getConcatDepName(c));
+                    return dep;
+                }).collect(Collectors.toList());
+    }
+
+    /**
+     * Gets vulnerabilities.
+     *
+     * @return all vulnerabilities
+     */
+    public static List<VulnerabilityDTO> getVulnerabilities() {
+        final List<DependencyDTO> dependencies = getDependencies();
+        final List<OpenSourceVulnerability> vulnerabilities = VulnerabilityStore
+                .getInstance().getVulnerabilities();
+
+        List<VulnerabilityDTO> vulnerabilityDTOS = new ArrayList<>();
+        for (OpenSourceVulnerability v : vulnerabilities) {
+            for (Affected affected : v.getAffected()) {
+                String depGroup = getDepGroup(affected);
+                String depName = getDepName(affected);
+                DependencyDTO dependencyDTO = dependencies.stream()
+                        .filter(d -> d.getGroup().equals(depGroup)
+                                && d.getName().equals(depName))
+                        .findFirst().orElse(null);
+                if (dependencyDTO != null) {
+                    VulnerabilityDTO vulnerabilityDTO = new VulnerabilityDTO(
```

Comment:
Should we add the same logic for getting the vuln id here like in V8 app? Or will be this provider also moved to the backend?